src/ebonite/api/pipelines.py

```diff
@@ -0,0 +1,158 @@
+            description: List of pipelines for given task
+          404:
+            description: Task with given id does not exist
+        """
+        task_id = request.args.get('task_id')
+        TaskIdValidator(task_id=task_id)
+        task = ebonite.meta_repo.get_task_by_id(task_id)
+        if task is not None:
+            return jsonify([pj.serialize(x) for x in ebonite.get_pipelines(task)]), 200
+        else:
+            raise ObjectWithIdDoesNotExist('Task', task_id)
+
+    @blueprint.route('/<int:id>', methods=['GET'])
+    def get_pipeline(id: int) -> Tuple[Response, int]:
+        """
+        Get pipeline with given id from metadata repository
+        ---
+        parameters:
+          - name: id
+            in: path
+            type: integer
+            description: Id of the pipeline meant to be retrieved
+        responses:
+          200:
+            description: Pipeline with given id
+          404:
+            description: Pipeline with given id doesn't exist
+        """
+        pipeline = ebonite.meta_repo.get_pipeline_by_id(id)
+        if pipeline is not None:
+            return jsonify(pj.serialize(pipeline)), 200
+        else:
+            raise ObjectWithIdDoesNotExist('Pipeline', id)
+
+    @blueprint.route('/<int:id>', methods=['PATCH'])
+    def update_pipeline(id: int):
+        """
+        Updates pipeline
+        ---
+        parameters:
+          - name: id
+            in: path
+            type: integer
+            required: true
+          - name: body
+            in: body
+            required: true
+            schema:
+              required:
+                - task_id
+              properties:
+                name:
+                  type: string
+                  required: false
+                  description: New name of the pipeline
+                author:
+                  type: string
+                  required: false
+                  description: New author of the pipeline
+                steps:
+                  type: object
+                  required: false
+                input_data:
+                  type: object
+                  required: false
+                output_data:
+                  type: object
+                  required: false
+                task_id:
+                  type: integer
+                  required: true
+                  description: task_id of the task pipeline belongs to
+        responses:
+          204:
+            description: Pipeline updated successfully
+          404:
+            description: Pipeline with given id does not exist
+          422:
+            description: Incorrect data sent in body
+        """
+        body = request.get_json(force=True)
+        body['id'] = id
+        old_pipeline = ebonite.meta_repo.get_pipeline_by_id(id)
+        if old_pipeline is None:
+            raise ObjectWithIdDoesNotExist('Pipeline', id)
+        # TODO: Should I leave options for this fields to be set through API
```

Comment:
maybe add an update method to EboniteParam (and make EboniteObject subclass of EboniteParam to also use this)
this update method will go through all fields (using pj helpers) and update them (either directly or using nested .update for nested EboniteParam instances)
I did not think this through for every possible case, just so you know :)